### ISSUE TITLE
event.target of panstart is not the actual target where pan started

### ISSUE THREAD

**Issue by LeJared** [hammerjs/hammer.js#815]
**Example**
http://codepen.io/anon/pen/doZKmr

When dragging one of the red squares the `event.target` of `panstart` ist displayed in the gray area below. Note, that  hammer is not bound to the red squares itself but to the outer div#stage!

**Expected behavior**
When dragging those red squares with the **_mouse_** I expect the `event.target` of `panstart` to be the red square where the user actually clicked (where the `mousedown` event had occurred).

**Actual behavior**
`event.target` is the element where the mouse pointer is, after the it has moved at least _pan.options.threshold_ pixels away from the `mousedown` position. On slow systems/browsers this also may be a much greater distance since the mouse may have moved much further away until the next mousemove-event is issued from the browser.

**Workaround**
My current workaround is, that I use the current pointer position of the mouse at `panstart`, subtract `event.deltaX` /  `event.deltaY` to compute the actual panstart-position and get the element from that point with `document.elementFromPoint(x, y)`

**Comment by runspired** [comment]
You can get the original target from  from the `e.srcEvent.target`.  You don't need the `document.elementFromPoint` hack. 

@arschmitz  I don't believe this is a bug, it was likely the original intention that `target` referred to the element being listened on (to differentiate manager instances), given that is where the corresponding gesture DOM event is originating.

I do think when improving the events created we ought to consider mirroring the source event's target.

Closing as I don't believe this is a bug, but I do think #808's eventing upgrade should this take into consideration.

**Comment by LeJared** [comment]
No, you can't use `e.srcEvent.target`. It does not change anything. see updated example:
http://codepen.io/anon/pen/XbzxwP

The problem ist not, that e.target is set to the wrong element. It is set at the *wrong time*. It is supposed to be the element where **mousedown** had occurred. Instead it is set to the element, where **mousemove** was fired when *pan.options.threshold* was reached and thus a pan was recognized and panstart fired.

**BTW:** e.target never was the element being listened to. It always was the actual element where the event was fired *within* the element you are listen to

**Comment by runspired** [comment]
@arschmitz When I tested this I missed that it was the mousemove not the mousedown, it definitely seems to me that the src event should be set to mousedown.

**Comment by runspired** [comment]
Related:
- #818
- #824

I believe this is fixed by: https://github.com/hammerjs/hammer.js/commit/92f2d76188480d967e738a19cd508d0b94a31329

**Comment by runspired** [comment]
Not fixed yet: http://codepen.io/runspired/pen/meaPgM
